lib/AmdMainTemplatePlugin.js

```diff
@@ -14,9 +14,13 @@ AmdMainTemplatePlugin.prototype.apply = function(compilation) {
 		var externals = chunk.modules.filter(function(m) {
 			return m.external;
 		});
-		var externalsDepsArray = JSON.stringify(externals.map(function(m) {
-			return typeof m.request === "object" ? m.request.amd : m.request;
-		}));
+		var externalsDepsArray = JSON.stringify(
+			externals.map(function(m) {
+				return typeof m.request === "object" ? m.request.amd : m.request;
+			}).filter(function(m) {
+				return !!m
+			})
+		);
 		var externalsArguments = externals.map(function(m) {
```

Comment:
ah, that one is going to be a little tricky. do you think the proper solution here is just to leave the `null`s in `externalsDepsArray`?